### DISCUSSION TITLE
Display actual exception class for ViewException

### DISCUSSION THREAD

**Discussion by jasonvarga** [facade/ignition#237] [Ideas] [answered]
When a ViewException is thrown, naturally the error page says ViewException.

![image](https://user-images.githubusercontent.com/105211/72458100-f2b7ad80-3795-11ea-947c-690186273085.png)

Would it be possible to replace that with the actual exception?

Of course, we can look in the stack trace to see which exception was really thrown, but this could be more useful.

**Comment by freekmurze** [comment]
Which exception would you expect here?

**Reply by jasonvarga** [reply]
I'd expect the error/exception that was originally thrown.

If I cause an error outside the view, I see my `SpecificException`:

```
Route::get('test', function () {
    throw new SpecificException('Oh no');
});
```
![image](https://user-images.githubusercontent.com/105211/87355532-89d7a500-c52e-11ea-8934-d541fc44a597.png)


If I cause the exception while in the view, I get a `ViewException`:

```
Route::get('/', function () {
    $foo = function () {
        throw new SpecificException('Oh no');
    };
    return view('welcome', ['foo' => $foo]);
});
```

```
{{ $foo() }}
```

![image](https://user-images.githubusercontent.com/105211/87355670-c1465180-c52e-11ea-9b0a-632ef57e8924.png)

There's no easy way to see that it was my `SpecificException`.

**Comment by freekmurze** [comment, marked as answer]
This has been fixed via 1.3.3 of flare-client-php

Running `composer update` will pull that version in and you'll see the actual exception class instead of the view error.

Enjoy!

**Reply by jasonvarga** [reply]
Fantastic, thank you!
